### PR TITLE
Request structure documentation 

### PR DESCRIPTION
**Pull request by Breigner01**
# Description

This PR adds some doc for the target member of the `Request` structure.
It also reorders the members so that it corresponds to the order it is in the HTTP request.
It also renames the `fields` field to `headers`.
For more explanations refer to the commits

# Breaking changes

The order change of the structure members might break some structure constructions which rely on the order of the structure members.
The name of the field `fields` changed to `headers`

# Checklist

- [ ] I have assigned this PR to myself
- [ ] I have added at least 1 reviewer
- [x] I have tested this code
- [x] I have added sufficient documentation and/or updated documentation to reflect my changes

### CODE REVIEW

File: include/ziapi/Http.hpp
Comment (rojasdiegopro) [inline include/ziapi/Http.hpp:28]: This member variable should be named `headers`

File: include/ziapi/Http.hpp
Comment (rojasdiegopro) [inline include/ziapi/Http.hpp:22]: I don't think ZIAPI should include documentation on the HTTP protocol. If you start explaining one thing from the protocol, you're bound to explaining everything. I believe that if a person is not familiar with the concept of a HTTP request "target" they should look it up themselves instead of relying on ZIAPI's documentation which might be untrue or incomplete.
Feel free to collect Gabriel's or Thibault's opinion on this.

**Review by rojasdiegopro — CHANGES_REQUESTED** [review]
Thanks for your submission! We're going to discuss this change internally.

In the meantime, here is a list of improvements for your PR:
- Update the fields of the `Response` structure, not just the `Request`
- Update the documentation to reflect the changes introduced in the PR

File: include/ziapi/Http.hpp
Comment (Emeraude) [reply include/ziapi/Http.hpp:22]: I don't think it can be qualified as documentation of the HTTP protocol. It's just an example to explicit which variables match which parts of the request.

As always, I'll let the maintainer decide, please don't just use my opinion as a "Thibaut told us to do it" joker :yum:

Comment (Ph4ntomas) [reply include/ziapi/Http.hpp:22]: I agree with Thibaut that this is in no way a documentation of the HTTP protocol.

> I believe that if a person is not familiar with the concept of a HTTP request "target" they should look it up themselves instead of relying on ZIAPI's documentation which might be untrue or incomplete.

I see two issues about this particular statement : 
- the fact that the documentation is not complete should not prevent it to actually exists. It could just be a quick remainder, with a link that redirect you to a better source of information.
- the actual definition in the RFC 2616 is "Resource-URI". As such, you cannot expect a user to "look it up", as it's not the proper terminology to begin with, and it only apply to your particular implementation. As `target` is not properly formalized, one could understand it as the `host [ ":" port ] [ abs_path]` part of the URI, without the query parameters + scheme, as they are not part of what actually is the "target" of the message, but the protocol used to communicate, as well as part of the actual message sent to that target

Now as Thibaut said, it's up to the maintainer to decide, I´m only expressing an opinion here

Comment (rojasdiegopro) [reply include/ziapi/Http.hpp:22]: Okay! Very valid feedback. Thanks!

**Comment by rojasdiegopro** [comment]
Let me know @Breigner01 if you need help with completing this PR. We'll merge as soon as you update the docs and the `Response` structure.

**Review by rojasdiegopro — APPROVED** [review]
Looks good to me!

**Pull request merged**
